python/lsst/ts/ATMCSSimulator/mcs_csc.py

```diff
@@ -576,223 +565,232 @@ def update_events(self):
         and for axes that have run into a limit switch, abort the axis,
         disable its drives and set its brakes.
         """
-        t = curr_tai()
-        curr_pos = np.array([actuator.curr.pva(t)[0] for actuator in self.actuators], dtype=float)
-        m3actuator = self.actuators[Axis.M3]
-        axes_in_use = set([Axis.Elevation, Axis.Azimuth, Axis.M3])
-
-        # Handle M3 actuator; set_cmd needs to be called to transition
-        # from slewing to tracking, and that is done here for M3
-        # (the trackPosition command does that for the other axes).
-        m3arrived = m3actuator.kind(t) == path.Kind.Slewing and t > m3actuator.curr[-1].t0
-        if m3arrived:
-            m3actuator.curr = path.Path(path.TPVAJ(t0=t, p0=m3actuator.cmd.p0), kind=path.Kind.Stopped)
-        exit_port, rot_axis = self.m3_port_rot(t)
-        if rot_axis is not None:
-            axes_in_use.add(rot_axis)
+        try:
+            tai = salobj.current_tai()
+            curr_pos = np.array([actuator.path.at(tai).position for actuator in self.actuators], dtype=float)
```

Comment:
I suggest `current_position` instead of `curr_pos`. This comment is general for the full PR.